From e0a6d3e6ff180ad19481427b7fe4b9a12e4001b1 Mon Sep 17 00:00:00 2001
From: "Wong, Vincent Por Yin" <vincent.por.yin.wong@intel.com>
Date: Wed, 1 Jul 2020 16:02:00 +0800
Subject: [PATCH 53/54] net: stmmac: skip hwts_rx_en check in stmmac_xsk

We shouldn't do a OR check with hwts_rx_en, as hwts_all
is required to timestamp generic packets.

Signed-off-by: Wong, Vincent Por Yin <vincent.por.yin.wong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c
index ffa793b46d21..b3ce8ad3bb13 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c
@@ -653,7 +653,7 @@ int stmmac_rx_zc(struct stmmac_priv *priv, int budget, u32 queue)
 
 		xdp.data = buf->umem_addr;
 
-		if (unlikely(priv->hwts_all) || unlikely(priv->hwts_rx_en)) {
+		if (unlikely(priv->hwts_all)) {
 			/* We use XDP meta data to store T/S */
 			xdp.data_meta = xdp.data - sizeof(ktime_t);
 
-- 
2.17.0

